rotkehlchen/data/importer.py

```diff
@@ -182,3 +186,183 @@ def import_cointracking_csv(self, filepath: Path) -> None:
+                # trades crypto to crypto
+                base_asset = Asset(to_currency)
+                quote_asset = Asset(currency)
+                if quote_asset is None:
+                    raise DeserializationError('Got a trade entry with an empty quote asset')
+                base_amount_bought = deserialize_asset_amount(to_amount)
+                quote_amount_sold = deserialize_asset_amount(amount)
+            else:
+                base_asset = Asset(currency)
+                quote_asset = Asset(native_currency)
+                base_amount_bought = deserialize_asset_amount(amount)
+                quote_amount_sold = deserialize_asset_amount(native_amount)
+
+            rate = Price(abs(quote_amount_sold / base_amount_bought))
+            pair = TradePair(f'{base_asset.identifier}_{quote_asset.identifier}')
+            trade = Trade(
+                timestamp=timestamp,
+                location=Location.CRYPTOCOM,
+                pair=pair,
+                trade_type=trade_type,
+                amount=base_amount_bought,
+                rate=rate,
+                fee=fee,
+                fee_currency=fee_currency,
+                link='',
+                notes=notes,
+            )
+            self.db.add_trades([trade])
+
+            # crypto_purchase is buying crypto assets using a card so
+            # it should also be considered as a deposit movement
+            if row_type == 'crypto_purchase':
+                asset_movement = AssetMovement(
+                    location=Location.CRYPTOCOM,
+                    category=AssetMovementCategory.DEPOSIT,
+                    timestamp=timestamp,
+                    asset=Asset(native_currency),
+                    amount=deserialize_asset_amount(native_amount),
+                    fee=fee,
+                    fee_asset=fee_currency,
+                    link='',
+                )
+                self.db.add_asset_movements([asset_movement])
+
+        elif row_type in (
+            'crypto_earn_program_created',
+            'lockup_lock',
+            'lockup_unlock',
+            'dynamic_coin_swap_bonus_exchange_deposit',
+            'crypto_wallet_swap_debited',
+            'crypto_wallet_swap_credited',
+            'lockup_swap_debited',
+            'lockup_swap_credited',
+            'dynamic_coin_swap_debited',
+            'dynamic_coin_swap_credited',
+            'dynamic_coin_swap_bonus_exchange_deposit',
+        ):
+            # those types are ignored because it doesn't affect the wallet balance
+            # or are not handled here
+            return
+        else:
+            raise UnsupportedCryptocomEntry(
+                f'Unknown entrype type "{row_type}" encountered during '
+                f'cryptocom data import. Ignoring entry',
+            )
+
+    # A swaping trade is a debited row + a credited row
+    def _import_cryptocom_swap(self, header: list, data: Any) -> None:
```

Comment:
Can you add a docstring for this function? Not 100% sure what it does?

It searches the entire CSV for `'Transaction Kind' == `'dynamic_coin_swap_debited'` and  `'Transaction Kind' == `'dynamic_coin_swap_credited'` and saves the **last** row where this happened.

Question here: Can't there be multiple rows that do that? Don't you need them all?

And then only if both exist you save some kind of coin swap trade. But I would expect that multiple of those can exist in the CSV. So I am not sure how the function find them all.

Reply:
I was thinking about a specific and recent swapping event here but you're right it should handle multiple swapping events. I'll fix that and add a docstring here 👍